microsoft/testsuites/dpdk/dpdkperf.py

```diff
@@ -0,0 +1,242 @@
+from typing import Any, Dict, Tuple
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+    simple_requirement,
+)
+from lisa.environment import Environment
+from lisa.features import Sriov
+from lisa.messages import NetworkPPSPerformanceMessage, create_message
+from lisa.tools import Lscpu
+from lisa.util import constants
+from microsoft.testsuites.dpdk.dpdkutil import (
+    DpdkTestResources,
+    verify_dpdk_send_receive,
+    verify_dpdk_send_receive_multi_txrx_queue,
+)
+
+
+@TestSuiteMetadata(
+    area="dpdk",
+    category="performance",
+    description="""
+    This test suite is to validate DPDK performance
+    """,
+)
+class DpdkPerformance(TestSuite):
+    TIMEOUT = 12000
+
+    @TestCaseMetadata(
+        description="""
+        This test case gathers performance data for dpdk send-recv-fwd: failsafe pmd
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2, network_interface=Sriov(), min_nic_count=2
+        ),
+    )
+    def perf_dpdk_failsafe_dual_core(
```

Comment:
Should it limit the environment has at least two cores?

Reply:
Nope, DPDK will bind to the cores you tell it to. Since they're committed to doing dpdk stuff you would want your system to have more than just those cores.

Reply:
Does it work with one core vm sizes?